controllers/system/system_controller.go

```diff
@@ -1125,30 +1125,33 @@ func (r *SystemReconciler) ReconcileRequired(instance *starlingxv1.System, spec
 		current.Certificates = &res
 	}
 
-	if spec.DeepEqual(current) {
-		logSystem.V(2).Info("no changes between spec and current configuration")
-		instance.Status.Delta = ""
-		return nil, false
-	}
-
-	logSystem.Info("spec is:", "values", spec)
-
-	logSystem.Info("current is:", "values", current)
+	inSync := spec.DeepEqual(current)
+	oldDelta := instance.Status.Delta
 
-	deltaString, err := common.GetDeltaString(current, spec, common.SystemProperties)
-	if err != nil {
+	if inSync {
+		instance.Status.Delta = ""
+	} else if deltaString, err := common.GetDeltaString(current, spec, common.SystemProperties); err == nil {
+		instance.Status.Delta = deltaString
+	} else {
 		logSystem.Info(fmt.Sprintf("failed to get Delta status:  %s\n", err))
 	}
 
-	if deltaString != "" {
-		logSystem.Info(fmt.Sprintf("delta configuration:%s\n", deltaString))
-		instance.Status.Delta = deltaString
+	if oldDelta != instance.Status.Delta {
+		logSystem.Info(fmt.Sprintf("delta configuration:%s\n", instance.Status.Delta))
 		err = r.Client.Status().Update(context.TODO(), instance)
 		if err != nil {
 			logSystem.Info(fmt.Sprintf("failed to update status:  %s\n", err))
 		}
 	}
 
+	if inSync {
```

Comment:
Not prefer to return because of insync, lin1155 should be enough to handle this case